src/Data/GADT/Internal.hs

```diff
@@ -63,14 +79,24 @@ instance GShow TR.TypeRep where
 --
 -- | >>> gshow (InL Refl :: Sum ((:~:) Int) ((:~:) Bool) Int)
 -- "InL Refl"
-instance (GShow a, GShow b) => GShow (Sum a b) where
+instance ( GShow a, GShow b
+#if __GLASGOW_HASKELL__ >= 806
+         , forall x. Show (a x)
+         , forall x. Show (b x)
+#endif
```

Comment:
i don't understand this. shouldn't `GShow a` imply `forall x. Show (a x)`, because it's `class forall x. Show (t x)) => GShow t`. Why GHC doesn't accept this?